src/Columns/ColumnVector.cpp

```diff
@@ -895,13 +919,14 @@ ColumnPtr ColumnVector<T>::replicate(const IColumn::Offsets & offsets) const
     }
 #endif
 
-    auto it = res->getData().begin(); // NOLINT
-    for (size_t i = 0; i < size; ++i)
-    {
-        const auto span_end = res->getData().begin() + offsets[i]; // NOLINT
-        for (; it != span_end; ++it)
-            *it = data[i];
-    }
+    /// This formula provides the optimum for a very simplified and probably wrong model for the number of additional checks (offsets[i] == offsets[i + window - 1])
```

Comment:
Interesting, can we change the format of offsets to optimize it unconditionally? E.g. we can convert offsets counts, and then RLE-encode zeroes (do it during offsets calculation).

Also, we can count statistics during building, and choose implementation based on the number of skipped rows.

Reply:
@Algunenano, maybe you'd find this interesting

Reply:
Another approach would be to make offset something like `{ UInt64 index, UInt64 quantity }` so during the generation you only push offsets with quantity >= 0. That way you don't have any branches inside the loop.

This uses 2x as much memory in the worse case (when quantity is 1), will use the same memory if 50% of elements have quantity = 0, and less memory if it's over that. Also, it removes the `offsets[i+1]-offsets[i]` which I've always find really confusing

Reply:
Another thought: Maybe we need different replicate() for different use cases.